core/src/dht/dht_reducers.rs

```diff
@@ -68,6 +78,32 @@ pub(crate) fn reduce_hold_entry(
     Some(new_store)
 }
 
+//
+pub(crate) fn reduce_add_crud_meta(
+    _context: Arc<Context>,
+    old_store: &DhtStore,
+    action_wrapper: &ActionWrapper,
+) -> Option<DhtStore> {
+    // Get Action input
+    let action = action_wrapper.action();
+    let (entry, _maybe_crud) = unwrap_to!(action => Action::Commit);
+    // Add crud-status metadata to local storage
+    let new_store = (*old_store).clone();
+    let meta_storage = &new_store.meta_storage().clone();
+    let status_eav = create_crud_status_eav(&entry.address(), CrudStatus::LIVE);
+    let res = (*meta_storage.write().unwrap()).add_eav(&status_eav);
+    if res.is_err() {
+        // TODO #439 - Log the error. Once we have better logging.
+        println!(
+            "reduce_add_crud_meta: meta_storage write failed!: {:?}",
+            res.err().unwrap()
+        );
+        return None;
+    }
+    // Done
+    Some(new_store)
+}
```

Comment:
This function can and should be called from `reduce_hold_entry`, instead of bound to `Action::Commit`.